domain/src/test/java/org/oppia/domain/analytics/AnalyticsControllerTest.kt

```diff
@@ -0,0 +1,291 @@
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createExplorationContext(
+        TEST_TOPIC_ID,
+        TEST_STORY_ID,
+        TEST_EXPLORATION_ID
+      )
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(EXPLORATION_CONTEXT)
+  }
+
+  @Test
+  fun testController_logClickEvent_withTopicContext_checkLogsEvent() {
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createTopicContext(TEST_TOPIC_ID)
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(TOPIC_CONTEXT)
+  }
+
+  @Test
+  fun testController_logClickEvent_withStoryContext_checkLogsEvent() {
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createStoryContext(TEST_TOPIC_ID, TEST_STORY_ID)
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(STORY_CONTEXT)
+  }
+
+  @Test
+  fun testController_logClickEvent_withNoContext_checkLogsEvent() {
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      null
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(ACTIVITYCONTEXT_NOT_SET)
+  }
+
+  @Test
+  fun testController_createExplorationContext_isSuccessful() {
```

Comment:
Instead of `isSuccessful`, lets make it more specific like `testController_createExplorationContext_returnsCorrectExplorationContext()`. Here and below

Reply:
Done.